map/src/App.tsx

```diff
@@ -120,6 +130,12 @@ class App extends React.Component<Props, State> {
     }));
   };
 
+  private setTranslate = (mutator: TranslateMutator) => {
+    this.setState(state => ({
+      translate: mutator(state.translate),
+    }));
+  };
+
```

Comment:
rather than providing a callback that allows components to change the language in the App component state, it would be better to make these requests go via i18n, and trigger an event to update the state in `App`